x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/package-info.java

```diff
@@ -0,0 +1,168 @@
+ * </p>
+ * <ol>
+ *     <li>
+ *         Copy an existing aggregator to use as a base. You'll usually make one per type. Check other classes to see the naming pattern.
+ *         You can find them in {@link org.elasticsearch.compute.aggregation}.
+ *         <p>
+ *             Note that some aggregators are autogenerated, so they live in different directories.
+ *             The base is {@code x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/}
+ *         </p>
+ *     </li>
+ *     <li>
+ *         Make a test for your aggregator.
+ *         You can copy an existing one from {@code x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/}.
+ *         <p>
+ *             Tests extending from {@code org.elasticsearch.compute.aggregation.AggregatorFunctionTestCase}
+ *             will already include most required cases. You should only need to fill the required abstract methods.
+ *         </p>
+ *     </li>
+ *     <li>
+ *         Check the Javadoc of the {@link org.elasticsearch.compute.ann.Aggregator}
+ *         and {@link org.elasticsearch.compute.ann.GroupingAggregator} annotations.
+ *         Add/Modify them on your aggregator.
+ *     </li>
+ *     <li>
+ *         The {@link org.elasticsearch.compute.ann.Aggregator} JavaDoc explains the static methods you should add.
+ *     </li>
+ *     <li>
+ *         After implementing the required methods (Even if they have a dummy implementation),
+ *         run the CsvTests to generate some extra required classes.
+ *         <p>
+ *             One of them will be the {@code AggregatorFunctionSupplier} for your aggregator.
+ *             Find it by its name ({@code <Aggregation-name><Type>AggregatorFunctionSupplier}),
+ *             and return it in the {@code toSupplier} method in your function, under the correct type condition.
+ *         </p>
+ *     </li>
+ *     <li>
+ *         Now, complete the implementation of the aggregator, until the tests pass!
+ *     </li>
+ * </ol>
+ *
+ * <h3>StringTemplates</h3>
+ * <p>
+ *     Making an aggregator per type may be repetitive. To avoid code duplication, we use StringTemplates:
+ * </p>
+ * <ol>
+ *     <li>
+ *         Create a new StringTemplate file.
+ *         Use another as a reference, like
+ *         {@code x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/X-TopValuesListAggregator.java.st}.
+ *     </li>
+ *     <li>
+ *         Add the template scripts to {@code x-pack/plugin/esql/compute/build.gradle}.
+ *         <p>
+ *             You can also see there which variables you can use, and which types are currently supported.
+ *         </p>
+ *     </li>
+ *     <li>
+ *         After completing your template, run the generation with {@code ./gradlew :x-pack:plugin:esql:compute:stringTemplates}.
+ *         <p>
+ *             You may need to tweak some import orders per type so they don't raise warnings.
```

Comment:
Sorry about that. We make the stringtemplated code pass linting and the import order is mighty annoying. Maybe we should turn off that warning for these. I find getting the import order right to be one of the most tedious things with string templated files.

Reply:
Just removed it, and added generated-src to spotless exclusions. I should have started with that instead

Reply:
Which makes me wonder if we want spotless for something there, as it wasn't excluded before. But as /generated/ is excluded, it feels weird to treat that generated-src differently

Reply:
After thinking again about this, restored it. As stringtemplated files are manually created, I guess it makes sense to run checks on them

Reply:
I think we *do* want to spotless them, but we might not want to run the import order bits. I think that's a change that wants debates so let's keep it separate.